### DISCUSSION TITLE
Using xr.apply_ufunc in a masked DataArray in an efficient way

### DISCUSSION THREAD

**Discussion by ricardobarroslourenco** [pydata/xarray#7627] [General] [answered]
This one is related to this other thread (https://github.com/pydata/xarray/discussions/7591).

I'm applying multiple regressions over lat,lon pairs using `apply_ufunc`, but I have noticed (by simply printing within my function) that the apply does not take in consideration the masking I have done in the data (so several values with the `where` clause are NaN's, such any water body for example). 

How should I deal with this situation? Is there a way to constrain the `apply_ufunc` only on valid data?

**Comment by dcherian** [comment, marked as answer]
No, that's not possible. You could short-circuit the all-NaN case in your custom function though.

So 

```

if np.isnan(x).all():
    return np.nan
```

Or something lilke that

**Reply by ricardobarroslourenco** [reply]
Thanks for the answer. It is good to know.
